mutation/mutation_cleaner.hh

```diff
@@ -24,6 +25,7 @@ class mutation_cleaner_impl final {
         snapshot_list snapshots;
         logalloc::allocating_section alloc_section;
         bool done = false; // true means the worker was abandoned and cannot access the mutation_cleaner_impl instance.
+        rwlock merging_enabled; // Allows for pausing the background merging. Used only for testing purposes.
```

Comment:
Is a rwlock really needed? Will you have multiple "reader"s?

In fact I'm not sure a lock is needed at all, since the non-test size only uses trylock. Isn't it enough to have a bool?

Reply:
Changed to an int (like a bool, but reentrant) in v2.